docs/code_standards.md

```diff
@@ -0,0 +1,160 @@
+We welcome issues and pull requests to help improve the xnat-tools model.  We use the following guidelines to ensure standards are met.
+
+## Workflow
+
+When working on xnat-tools, we using a flow a simple flow based on following rules:
+
+1. Use topic/feature branches, no direct commits on master.
+2. Perform tests and code reviews before merges into master, not afterwards.
+3. Everyone starts from master, and targets master.
+4. Commit messages reflect intent.
+
+### Branches
+
+* `master` is the default branch and where releases are made off. This branch should be in clean/working conditions at all times. This branch is protected and can only be merged from Pull Requests for topic branches
+* topic branches are created for new features, fixes, or really any changes
+
+### Comment styles
+
+Developers should use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/). [Commitizen](https://commitizen-tools.github.io/commitizen/) comes installed as a `dev` dependency via poetry and you can use it to help you format your commit messages. It can also be installed as stand alone CLI or as a plugin on VSCODE.
+
+We use Commitizen in GitHub Actions to automatically bump the version and update the [CHANGELOG.md](CHANGELOG.md) of this package according to the commit messages. Using conventional commits guarantees than Commitizen works appropriately.  
+
+## Testing
+
+In the future we will add testing as part of CI. For the moment, you'll need to test yourself
+
+`pytest` is the library used for testing.
+
+To run all of the tests:
+```
+poetry run pytest -x -s -o log_cli=true --log-cli-level=INFO
+```
+
+To run only a file
+
+
+```
+poetry run pytest -x -s -o log_cli=true --log-cli-level=INFO tests/integration/test_export_typer.py
+```
+
+`-s` makes sure that `stdout` is printed to terminal
+`-o log_cli=true --log-cli-level=INFO` allows the logger output to go to cli
+`-x` exists on first failure
+
+You will need to have a local `.env` file where you set some environment variables, i.e. `XNAT_PASS`
+
+At the moment the tests can not run bids validation. To do so, you can comment out the line that cleans the output directory and run the validator manually using docker. For instance
+
+```
+bids_directory=${PWD}/tests/xnat2bids/ashenhav/study-1222/bids/
+docker run -ti --rm -v ${bids_directory}:/data:ro bids/validator /data
+```
+
+
+## Code Style
+
+### Typing
+
+Please use [type hints](https://mypy.readthedocs.io/en/stable/) on all signatures where reasonable.  This will make sure the code is more readable, can be statically tested for type soundness, and helps fill in the documentation.
+
+Run the below to check for type errors:
+```
+mypy xnat_tools
```

Comment:
Should this be a part of the `.pre-commmit.yaml` checks?

Reply:
It is now, I'll update the docs to make it clear